src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -525,45 +529,59 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
     } getOrElse Future.successful(())
   }
 
-
   private def validateClusterRequestBucketObjectUri(userEmail: WorkbenchEmail, googleProject: GoogleProject, clusterRequest: ClusterRequest)(implicit executionContext: ExecutionContext): Future[Unit] = {
-    for {
+    val transformed = for {
+      // Get a pet token from Sam. If we can't get a token, we won't do validation but won't fail cluster creation.
+      petToken <- OptionT(serviceAccountProvider.getAccessToken(userEmail, googleProject))
+
+      // Validate the user script URI
       _ <- clusterRequest.jupyterUserScriptUri match {
-        case Some(userScriptUri) => validateBucketObjectUri(userEmail, googleProject, userScriptUri.toUri)
-        case None => Future.successful(())
+        case Some(userScriptUri) => OptionT.liftF[Future, Unit](validateBucketObjectUri(userEmail, petToken, userScriptUri.toUri))
+        case None => OptionT.pure[Future, Unit](())
       }
 
+      // Validate the extension URIs
       _ <- clusterRequest.userJupyterExtensionConfig match {
         case Some(config) =>
           val extensionsToValidate = (config.nbExtensions.values ++ config.serverExtensions.values ++ config.combinedExtensions.values).filter(_.startsWith("gs://"))
-          Future.traverse(extensionsToValidate)(x => validateBucketObjectUri(userEmail, googleProject, x))
-        case None => Future.successful(())
+          OptionT.liftF(Future.traverse(extensionsToValidate)(x => validateBucketObjectUri(userEmail, petToken, x)))
+        case None => OptionT.pure[Future, Unit](())
       }
-    } yield()
+    } yield ()
+
+    transformed.value.void
```

Comment:
Sure. Basically because of how `OptionT` works, ` transformed.value` returns `Future[Option[Unit]]`. We really don't care about the return value though, hence the `.void`.